game-core/src/main/java/games/strategy/engine/framework/startup/ui/panels/main/game/selector/GameFileSelector.java

```diff
@@ -35,11 +36,7 @@
       fileDialog.setDirectory(ClientSetting.saveGamesFolderPath.getValueOrThrow().toString());
       fileDialog.setFilenameFilter((dir, name) -> GameDataFileUtils.isCandidateFileName(name));
       fileDialog.setVisible(true);
-      final String fileName = fileDialog.getFile();
-      final String dirName = fileDialog.getDirectory();
-      return Optional.ofNullable(fileName)
-          .map(name -> new File(dirName, fileName))
-          .map(this::mapFileResult);
+      return Arrays.stream(fileDialog.getFiles()).findAny().map(this::mapFileResult);
```

Comment:
I think this needs documentation to make explicit that we expect 'getFiles()' to have 0 or 1 elements. Second, did you test this out to be sure it works and you get 0 values back if you type in a file that DNE?

Reply:
I wonder if this is a case where it's worth having Mac OS specific code? The lack of mac skinning would not ideal, but I don't think we can readily test this otherwise. If we can guarantee that a selected file exists or none is chosen, then we can probably simplify the callback logic and remove the 'save-game-does-not-exist' warning message (that would be dead code otherwise, as-is in this update, it's conditionally dead-code depending on the OS (which is really not ideal)).

Reply:
> I think this needs documentation to make explicit that we expect 'getFiles()' to have 0 or 1 elements.

Good point.

> Second, did you test this out to be sure it works and you get 0 values back if you type in a file that DNE?

Yes, no doubt there.

> I wonder if this is a case where it's worth having Mac OS specific code?

If you ask me it's silly using a different kind of dialog in the code for a specific OS just to get the skinning right.
I'm not even sure if JFileDialog looks different with the MacOS L&F, I could check that some time next week or something.
In case it doesn't look different because the L&F improved: Great, we can remove the code in the whole codebase.
In case it does look different we can ask the mac folks in the community how they feel about it. In case they have strong feelings about keeping it, we could create our own wrapper class that provides a higher level API so we don't have double logic everywhere.

But those kind of questions are definitely out of scope for this PR.